js/jsmind.screenshot.js

```diff
@@ -103,19 +112,23 @@
 
         shoot:function(){
             this.init();
-            this._draw();
-            this._watermark();
-            this._download();
-            this.clean();
+            var jsm = this;
+            this._draw(function(){
+                jsm._watermark();
+                jsm._download();
+                jsm.clean();
+            });
         },
 
         shootAsDataURL:function(){
             this.init();
-            this._draw();
-            this._watermark();
-            var url = this.canvas_elem.toDataURL();
-            this.clean();
-            return url;
+            var jsm = this;
+            this._draw(function() {
+                this._watermark();
+                var url = jsm.canvas_elem.toDataURL();
+                jsm.clean();
+                return url;
```

Comment:
True. Probably the only way to return this value is using a callback function. What you think?
Also, I didn't found a use of shootAsDataURL()